scripts/main/album.js

```diff
@@ -171,63 +204,101 @@ album.load = function (albumID, refresh = false) {
 		}, waitTime);
 	};
 
-	api.post(
-		"Album::get",
-		params,
-		function (data) {
-			processData(data);
-
-			tabindex.makeFocusable(lychee.content);
-
-			if (lychee.active_focus_on_page_load) {
-				// Put focus on first element - either album or photo
-				first_album = $(".album:first");
-				if (first_album.length !== 0) {
-					first_album.focus();
-				} else {
-					first_photo = $(".photo:first");
-					if (first_photo.length !== 0) {
-						first_photo.focus();
-					}
+	/**
+	 * @param {Album} data
+	 */
+	const successHandler = function (data) {
+		processAlbum(data);
+
+		lychee.content.show();
+		lychee.footer_show();
+		tabindex.makeFocusable(lychee.content);
+
+		if (lychee.active_focus_on_page_load) {
+			// Put focus on first element - either album or photo
+			const first_album = $(".album:first");
+			if (first_album.length !== 0) {
+				first_album.focus();
+			} else {
+				const first_photo = $(".photo:first");
+				if (first_photo.length !== 0) {
+					first_photo.focus();
 				}
 			}
-		},
-		null,
-		function (jqXHR) {
-			if (jqXHR.status === 403) {
-				password.getDialog(albumID, function () {
-					params.password = password.value;
-
-					api.post("Album::get", params, function (_data) {
-						albums.refresh();
-						processData(_data);
-					});
-				});
-				return true;
-			}
+		}
+
+		if (albumLoadedCB) albumLoadedCB(true);
+	};
+
+	/**
+	 * @param {XMLHttpRequest} jqXHR
+	 * @param {Object} params the original JSON parameters of the request
+	 * @param {?LycheeException} lycheeException the Lychee exception
+	 * @returns {boolean}
+	 */
+	const errorHandler = function (jqXHR, params, lycheeException) {
+		if (jqXHR.status !== 401 && jqXHR.status !== 403) {
+			// Any other error then unauthenticated or unauthorized
+			// shall be handled by the global error handler.
 			return false;
 		}
-	);
-};
 
-album.parse = function () {
-	if (!album.json.title) album.json.title = lychee.locale["UNTITLED"];
+		if (lycheeException.message.includes("Password required")) {
+			// If a password is required, then try to unlock the album
+			// and in case of success, try again to load album with same
+			// parameters
+			password.getDialog(albumID, function () {
+				album.load(albumID, albumLoadedCB);
```

Comment:
```suggestion
				albums.refresh();
				album.load(albumID, albumLoadedCB);
```

Unlocking a password-protected album can make visible the previously inaccessible cover photos, so we should ensure that the albums view is refreshed.